learning/tour-of-beam/frontend/lib/pages/tour/state.dart

```diff
@@ -103,27 +106,28 @@ class TourNotifier extends ChangeNotifier with PageStateMixin<void> {
   }
 
   Future<void> _onAuthChanged() async {
-    await _updateUnitProgressAndSnippet();
+    await _updateSnippetTypeSelector();
+    if (_snippetType != SnippetType.saved
```

Comment:
// If the user just signed in, preserve their local changes.
// If they will continue typing, this snippet will overwrite the cloud one.
// Otherwise if they navigate away and return to the unit,
// the cloud saved version will show.

Only make this preservation work one-way: from unauthenticated to authenticated.